### ISSUE TITLE
False positive non-iterable error for IntEnum

### ISSUE THREAD

**Issue by artoonie** [PyCQA/pylint#1539]
### Steps to reproduce
```python
import enum                                                                         
                                                                                    
class Numbers(enum.IntEnum):                                                        
    One = 0                                                                         
    Two = 1                                                                         
    Three = 2                                                                       
                                                                                    
class Letters(enum.Enum):                                                           
    A = 0                                                                           
    B = 1                                                                           
    C = 2                                                                           
                                                                                    
# pylint emits an error                                                             
for num in Numbers:                                                                 
    print num                                                                       
                                                                                    
# pylint is happy                                                                   
for let in Letters:                                                                 
    print let
```

### pylint --version
No config file found, using default configuration
pylint 1.8.0,
astroid 1.5.2
Python 2.7.13 (default, Dec 17 2016, 23:03:43)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]

**Comment by PCManticore** [comment]
This is fixed in the master branch of astroid.
